projects/observability/src/shared/dashboard/widgets/charts/cartesian-widget/interactions/cartesian-explorer-context-menu/cartesian-explorer-context-menu.component.ts

```diff
@@ -38,16 +43,21 @@ export class CartesianExplorerContextMenuComponent<TData> {
 
   public constructor(
     @Inject(POPOVER_DATA) data: CartesianSelectedData<TData>,
-    private readonly cartesainExplorerNavigationService: CartesainExplorerNavigationService
+    private readonly cartesainExplorerNavigationService: CartesainExplorerNavigationService,
+    private readonly timeRangeService: TimeRangeService
   ) {
     this.selectionData = data;
   }
 
   public menuSelectHandler = (_menu: ContextMenu): void => {
-    this.cartesainExplorerNavigationService.navigateToExplorer(
-      this.selectionData.timeRange.startTime,
-      this.selectionData.timeRange.endTime
-    );
+    if (_menu.name === 'Explore') {
+      this.cartesainExplorerNavigationService.navigateToExplorer(
+        this.selectionData.timeRange.startTime,
+        this.selectionData.timeRange.endTime
+      );
+    } else {
+      this.timeRangeService.toQueryParams(this.selectionData.timeRange.startTime, this.selectionData.timeRange.endTime);
```

Comment:
This should be calling set time range. There's an existing thread asking to fix the new method `toQueryParams` to **not** modify the global time range, it should only be producing a param object.